src/Chapter1.hs

```diff
@@ -640,8 +652,10 @@ Implement a function that returns the first digit of a given number.
 You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n
+    | abs n <= 9 = abs n
+    | otherwise = firstDigit (div (abs n) 10)
```

Comment:
Wonderful! 👌🏼 

Here, you can also move `abs n` into a variable in the `where` construction 🙂

Reply:
Would you recommend `where or `let`?
What is considered more idiomatic/what are the benefits?

Reply:
I personally prefer `where` as it allows me to first look at the wider picture, only then on the details in `where`.

`where` and `let` have different scoping boundaries, so you need to be careful and use the appropriate constructions.
In here, you can not write `let` before `guards`, so you would need to add it in the case itself, but that would mean that it works only inside that scope and won't be visible in another guard case.
`where` on the other hand, would be visible in scopes of both cases. 🙂 

This is not an easy topic to explain all details of the difference between `let` and `where`, so we decided not to include it in the course itself. But I hope the above makes some sense. You can ask any questions and we could discuss these constructions and scoping in details if you are interested 👌🏼

Reply:
I ended up using both. `where` for this one and `let` for the other.

Reply:
That is very interesting that `let` would work on a guard-per-guard basis